### ISSUE TITLE
False negative: EI_EXPOSE_REP not reported when getter is defined in an anonymous class

### ISSUE THREAD

**Issue by Emilyaxe** [spotbugs/spotbugs#4032]
**Describe the bug**
SpotBugs flags `EI_EXPOSE_REP` for a named nested class returning an internal mutable array, but misses the identical case in an anonymous class implementing the same interface. Since both return the same mutable `String[]` reference, the detector should not behave differently based on whether the class is named or anonymous.

**To Reproduce**

```java
// BEFORE — EI_EXPOSE_REP is reported (correct)
package demo.before;

public class Outer {
    private static final String[] names = new String[] {"a", "b"};

    public static Provider getProvider() {
        return new ProviderImpl();
    }

    public interface Provider {
        String[] getNames();
    }

    public static class ProviderImpl implements Provider {
        @Override
        public String[] getNames() {
            return names; // flagged: EI_EXPOSE_REP
        }
    }
}
```

```java
// AFTER — no warning reported (false negative)
package demo.after;

public class Outer {
    private static final String[] names = new String[] {"a", "b"};

    public static Provider getProvider() {
        return new Provider() {
            @Override
            public String[] getNames() {
                return names; // NOT flagged, but equally exposes internal state
            }
        };
    }

    public interface Provider {
        String[] getNames();
    }
}
```

Run:

```
spotbugs -textui -effort:max Outer.class
```

**Expected behavior**
`EI_EXPOSE_REP` should be reported on `getNames()` in the anonymous class in `AFTER.java`, matching the behavior for the named nested class in `BEFORE.java`.

**Actual behavior**
No violation is reported for the anonymous-class version, although it exposes the internal `names` array identically.

**Environment**

- Java version: JDK 21 (Eclipse Temurin)
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI
